akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -538,33 +538,37 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
   }
 
   private def runInboundStreams(): Unit = {
-    runInboundControlStream()
-    runInboundOrdinaryMessagesStream()
+    val noCompression = new NoInboundCompression(system)
+    val compression = createInboundCompressionTable(this)
+
+    runInboundControlStream(noCompression)
+    runInboundOrdinaryMessagesStream(compression)
     if (largeMessageDestinationsEnabled) {
       runInboundLargeMessagesStream()
     }
   }
 
-  private def runInboundControlStream(): Unit = {
+  private def runInboundControlStream(compression: InboundCompression): Unit = {
```

Comment:
btw, the split between InboundCompression and OutboundCompression was a sanity life saver